html_utils/utils.py

```diff
@@ -0,0 +1,21 @@
+from bs4 import BeautifulSoup
+
+def get_title(soup: BeautifulSoup):
+    # return the title of the page of empty string if not found
+    title = soup.find("title")
+    return title.string if title else None
+
+def get_meta_name(soup: BeautifulSoup):
+    # finds the meta tag and returns the content of the name attribute or empty string if not found
+    meta_name = soup.find("meta", {"name": "description"})
+    return meta_name["content"] if meta_name else None
+
+def get_favicon_url(soup: BeautifulSoup):
+    # finds the favicon url or empty string if not found
+    favicon_url = soup.find("link", {"rel": "icon"})
+    return favicon_url["href"] if favicon_url else None
```

Comment:
same as previous comment. Another suggestion is to setup a default value: `favicon_url = None` then set `favicon_url` with the value of the search. Return `favicon_url`. You save an `if` statement and the code is easier to read.

Reply:
alternative added